### ISSUE TITLE
mj-text creates new html tag if it encounter <something inside

### ISSUE THREAD

**Issue by Pharmasolin** [mjmlio/mjml#2291]
**Describe the bug**

**To Reproduce**
1. Create a file with this MJML code: `<mj-text>
          If any small accent gems (<$100 value) break or fall out of their settings in the first year after we create your jewelry, we’ll replace the accent gems free of charge.
        </mj-text>`
2. HTML will look like `<td align="center" style="font-size:0px;padding:0word-break:break-word;">
                                <div style="font-family:Helvetica, Arial, sans-serif;font-size:13px;">If any small accent gems (<$100 value) break or fall out of their settings in the first year after we create your jewelry, we’ll replace accent gems free charge.< div>
                                  </$100>
                                </div>
                              </td>`

**Expected behavior**
I was expecting mj-text component would escape that character and wouldn't create this issue.

**MJML environment (please complete the following information):**
 - OS: Linus
 - MJML Version 4.9.3
 - CLI

**Screenshots**
Bug:
![image](https://user-images.githubusercontent.com/13041082/121884638-d36ed800-cd1b-11eb-950a-54404b1e66e0.png)
Expected behavior:
![image](https://user-images.githubusercontent.com/13041082/121884715-eb465c00-cd1b-11eb-8c0f-301b067dde25.png)

**Comment by iRyusa** [comment]
This works fine in try-it-live tho https://mjml.io/try-it-live/DdGPSUkMsWt Might be because you're using mj-html-attributes or inline styles at some point ? Best way to avoid that would be to encode `<` as `&lt;`

**Comment by Pharmasolin** [comment]
@iRyusa yep, I do both. Is it worth to add a note in docs?

**Comment by iRyusa** [comment]
Yup 👍  might need to test if it happens on both or only html-attributes (I would expect only on html-attributes). 

It might be hard to explain because it might happens with anything related to templating too 🤔

**Comment by kmcb777** [comment]
I confirmed that this causes problems with each of these, even without the others :
- mj-html-attributes => all following text is considered as html attributes and turned into `break="" or="" fall="" ...`
- inline-style => an extra "< mj-text>" is displayed at the end of the text
- minify option => crash
However we can't escape the text, otherwise some templating formats would be unusable.
I made a PR for adding this information in the doc, do you think this explanation is fine ? https://github.com/mjmlio/mjml/pull/2402

**Comment by dazgreer** [comment]
Closing this issue as the behaviour and 'fix' has been documented here: https://documentation.mjml.io/#ending-tags in the PR: https://github.com/mjmlio/mjml/pull/2402

Relating to template tags specifically, we have an issue: https://github.com/mjmlio/mjml/issues/3112 and PR: https://github.com/mjmlio/mjml/pull/3121 working on this
